notebook/notebookapp.py

```diff
@@ -1742,7 +1743,7 @@ def init_terminals(self):
 
         try:
             from .terminal import initialize
-            initialize(self.web_app, self.notebook_dir, self.connection_url, self.terminado_settings)
+            initialize(self.web_app, self.notebook_dir, self.connection_url, self.terminado_settings, self)
```

Comment:
Looking at this closer, this function signature bothers me. This is a server extension, so this method signature *should be* equivalent to `load_jupyter_server_extension(nbapp)`.  All of these arguments are attributes of `self`, so why pass them individually.

Reply:
Yep - I agree.  This was probably during a quick POC and never revisited. Thank you.